build/create-glossary-usages.js

```diff
@@ -3,33 +3,48 @@
  * -> for each glossary item (find references in each rule)
  * -> this is saved in `_data` which is later used in `pages/glossary`
  */
+const { config } = require('../package.json')
 const regexps = require('../utils/reg-exps')
 const createFile = require('../utils/create-file')
 const getAllMatchesForRegex = require('../utils/get-all-matches-for-regex')
-const getRulesMarkdownData = require('../utils/get-rules-markdown-data')
+const getMarkdownData = require('../utils/get-markdown-data')
 
-const init = async () => {
-	/**
-	 * Get all rules `markdown` data
-	 */
-	const rulesData = getRulesMarkdownData()
-
-	/**
-	 * Eg:
-	 * {
-	 *  `non-empty`: [
-	 *    { name: `aria valid ...`, slug: `rules/XXXXX` },
-	 *    ....
-	 *  ]
-	 *  ....
-	 * }
-	 */
-	const glossaryUsages = {}
+/**
+ * Get usage meta data
+ * @param {Object} frontmatter markdown frontmatter
+ * @param {String} type key denoting type/ grouping of markdown data, to help parse frontmatter
+ * @returns {Object|null}
+ */
+const getUsageMetaData = (frontmatter, type) => {
+	switch (type) {
+		case 'rule':
+			const { id, name } = frontmatter
+			return {
+				name: name,
+				slug: `rules/${id}`
+			}
+		case 'glossary':
+			const { key } = frontmatter
+			return {
+				key: `#${key}`
+			}
+		default:
+			return
+	}
+}
 
-	rulesData.forEach(ruleData => {
-		const { frontmatter, body } = ruleData
-		const { id: ruleId, name: ruleName, accessibility_requirements: ruleAccessibilityRequirements } = frontmatter
+/**
+ * Parse and find a given markdown content for references to glossary definitions
+ * 
+ * @param {Array<Object>} markdownData collection of parsed markdown data
+ * @param {String} type key denoting type/ grouping of markdown data, to help parse frontmatter
+ * @returns {Array<Object>}
+ */
+const getGlossaryUsagesInMarkdownData = (markdownData, type = 'rule') => {
```

Comment:
Can you please stop using "glossary" as a synonym for "definition". The glossary is a list of definitions. Each rule has one glossary, containing one or more definitions. There are no glossaries inside of other glossaries. Some definitions have references to other definitions.